File: custom_app/custom_app/web_form/issue/issue.py
import frappe

def get_context(context):
    # Get the session user
    session_user = frappe.session.user

    # Add the session user to the context dictionary
    context.user = session_user

    # Example: You can also set a default value for the User field
    context.default_user = session_user

    # Find the project associated with the session user
    project = frappe.get_all("S Project", filters={"customer_mail_id": session_user}, fields=["name", "customer_mail_id","customer"])
	# frappe.mgsprint(project)
    if project:
		# context.customer = project.customer
        # Get the customer_mail_id from the project
        context.project_name =", ".join([i.get("name") for i in project])
        context.customer_name =", ".join([i.get("customer") for i in project])
		# print()
    else:
        # Set a default value if the project is not found
        context.project_name = None
        context.customer_name = None

    # You can do more with the context as needed
    # ...

File: custom_app/custom_app/web_form/issue/issue.js
// frappe.ready(function() {
// 	// Example: Set the session user to the user field when the page is ready
// 	var sessionUser = "{{ user }}";
// 	$('input[data-fieldname="user"]').val(sessionUser);

// 	var project_name = "{{ project_name }}";
// 	alert(project_name);
// 	$('input[data-fieldname="project"]').val(project_name);
// 	// $('select[data-fieldname="project"]').val(project_name);
// 	// $('select[data-fieldname="project"]').trigger('change');


// 	// Example: Alert the session user when the page is ready
// 	// alert("Hello, " + sessionUser + "! The page is ready.");

// 	// You can bind events, manipulate the DOM, or perform other actions here
// });


frappe.ready(function() {
    
    // Example: Set the session user to the user field when the page is ready
    var sessionUser = "{{ user }}";
    $('input[data-fieldname="user"]').val(sessionUser);

    var customerName = "{{ customer_name }}"; // Use a single variable for customer name

    // Set the customer name in the input field
    $('input[data-fieldname="customer"]').val(customerName);

    var projectNames = "{{ project_name }}".split(','); // Split the project names by comma
    var projectDropdown = $('select[data-fieldname="project"]');

    // Clear existing options in the dropdown
    projectDropdown.empty();

    // Add each project name as an option to the dropdown
    for (var i = 0; i < projectNames.length; i++) {
        var projectName = projectNames[i].trim(); // Remove leading/trailing whitespaces
        projectDropdown.append('<option value="' + projectName + '">' + projectName + '</option>');
    }
    // You can bind events, manipulate the DOM, or perform other actions here
});
